src/components/CustomerQuota/CustomerQuotaScreen.tsx

```diff
@@ -145,11 +144,74 @@ export const CustomerQuotaScreen: FunctionComponent<CustomerQuotaProps> = ({
     if (cartState === "FETCHING_QUOTA") {
       const message =
         error.message ?? "Encounted an error while fetching quota";
-      showAlert(message, onCancel);
+      showAlert({
+        ...systemAlertProp,
+        description: message,
+        onOk: () => clearError()
+      });
     } else if (cartState === "DEFAULT" || cartState === "CHECKING_OUT") {
-      showAlert(error.message, () => clearError());
+      switch (error.message) {
+        case ERROR_MESSAGE.MISSING_SELECTION:
+          showAlert({
+            ...incompleteEntryAlertProp,
+            description: ERROR_MESSAGE.MISSING_SELECTION,
+            onOk: () => clearError()
+          });
+          break;
+
+        case ERROR_MESSAGE.MISSING_IDENTIFIER_INPUT:
+          showAlert({
+            ...incompleteEntryAlertProp,
+            description:
+              campaignFeatures?.campaignName === "TT Token"
+                ? ERROR_MESSAGE.MISSING_POD_INPUT
+                : campaignFeatures?.campaignName.includes("Vouchers")
+                ? ERROR_MESSAGE.MISSING_VOUCHER_INPUT
+                : ERROR_MESSAGE.MISSING_IDENTIFIER_INPUT,
+            onOk: () => clearError()
+          });
+          break;
+
+        case ERROR_MESSAGE.SERVER_ERROR:
+          showAlert({
+            ...systemAlertProp,
+            description: error.message,
+            onOk: () => clearError()
+          });
+          break;
+
+        case ERROR_MESSAGE.DUPLICATE_IDENTIFIER_INPUT:
+          showAlert({
+            ...systemAlertProp,
+            title: "Already used",
+            description:
+              campaignFeatures?.campaignName === "TT Token"
+                ? ERROR_MESSAGE.DUPLICATE_POD_INPUT
+                : ERROR_MESSAGE.DUPLICATE_IDENTIFIER_INPUT,
+            onOk: () => clearError()
+          });
+          break;
+
+        case ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT:
+          showAlert({
+            ...wrongFormatAlertProp,
+            description:
+              campaignFeatures?.campaignName === "TT Token"
+                ? ERROR_MESSAGE.INVALID_POD_INPUT
+                : ERROR_MESSAGE.INVALID_IDENTIFIER_INPUT,
+            onOk: () => clearError()
+          });
+          break;
+
+        default:
+          showAlert({
```

Comment:
Do you think a sentry error here will be useful so we know there's an unhandled error? Or should we just throw and let the error boundary take care of it?

Reply:
alright! will be throwing to error boundary!